src/libstore/meson.build

```diff
@@ -0,0 +1,458 @@
+  meson.project_build_root() / '__nothing_hardlink',
+  check : false,
+).returncode() == 0
+run_command('rm', '-f',
+  meson.project_build_root() / '__nothing_symlink',
+  meson.project_build_root() / '__nothing_hardlink',
+  check : true,
+)
+summary('can hardlink to symlink', can_link_symlink, bool_yn : true)
+configdata += { 'CAN_LINK_SYMLINK': can_link_symlink.to_int() }
+
+# Check for each of these functions, and create a define like `#define HAVE_LCHOWN 1`.
+#
+# Only need to do functions that deps (like `libnixutil`) didn't already
+# check for.
+check_funcs = [
+  # For canonicalising files from the build
+  'lchown',
+]
+foreach funcspec : check_funcs
+  define_name = 'HAVE_' + funcspec.underscorify().to_upper()
+  define_value = cxx.has_function(funcspec).to_int()
+  configdata += {
+    define_name: define_value,
+  }
+endforeach
+
+has_acl_support = cxx.has_header('sys/xattr.h') \
+  and cxx.has_function('llistxattr') \
+  and cxx.has_function('lremovexattr')
+configdata += {
+  'HAVE_ACL_SUPPORT': has_acl_support.to_int()
+}
+
+# Conditional to work around https://github.com/mesonbuild/meson/issues/13293
+if host_machine.system() != 'windows' and cxx.get_id() == 'gcc'
+  deps_private += dependency('threads')
+endif
+
+boost = dependency(
+  'boost',
+  required : true,
+  modules : ['container'],
+)
+# Actually private, but wrong type of dep for pkg config
+deps_other += boost
+
+curl = dependency('libcurl', 'curl', required : true)
+deps_private += curl
+
+# seccomp only makes sense on Linux
+is_linux = host_machine.system() == 'linux'
+seccomp_required = get_option('seccomp-sandboxing')
+if not is_linux and seccomp_required.enabled()
+  warning('Force-enabling seccomp on non-Linux does not make sense')
+endif
+seccomp = dependency('libseccomp', 'seccomp', required : seccomp_required, version : '>=2.5.5')
+if is_linux and not seccomp.found()
+  warning('Sandbox security is reduced because libseccomp has not been found! Please provide libseccomp if it supports your CPU architecture.')
+endif
+configdata += {
+  'HAVE_SECCOMP': seccomp.found().to_int(),
+}
+deps_private += seccomp
+
+nlohmann_json = dependency(
+  'nlohmann_json',
+  version : '>= 3.9',
+  required : true,
+)
+deps_public += nlohmann_json
+
+sqlite = dependency('sqlite3', 'sqlite', version : '>=3.6.19', required : true)
+deps_private += sqlite
+
+
+enable_embedded_sandbox_shell = get_option('embedded-sandbox-shell')
+if enable_embedded_sandbox_shell
+  # This one goes in config.h
+  # The path to busybox is passed as a -D flag when compiling libstore.
+  # Idk why, ask the old buildsystem.
+  configdata += {
+    'HAVE_EMBEDDED_SANDBOX_SHELL': 1,
+  }
+endif
+
+generated_headers = []
+foreach header : [ 'schema.sql', 'ca-specific-schema.sql' ]
+  generated_headers += custom_target(
+    command : [ 'bash', '-c', 'echo \'R"__NIX_STR(\' | cat - @INPUT@ && echo \')__NIX_STR"\'' ],
+    input : header,
+    output : '@PLAINNAME@.gen.hh',
+    capture : true,
```

Comment:
If you are already running a bash command, why run `capture: true`? This just means that meson will wrap the entire bash command inside a python wrapper script that is in charge of passing inputs/outputs.

What you really want is pretty simple:

```meson
    command: ['bash', '-c', 'echo \'R"__NIX_STR(\' | { cat - @INPUT@ && echo \')__NIX_STR"\'; } > "$1"', '_ignored_argv0', '@OUTPUT@'],